xdsl/dialects/pdl.py

```diff
@@ -539,9 +542,13 @@ class TypesOp(IRDLOperation):
     constantTypes: OptOpAttr[AnyArrayAttr]
     result: Annotated[OpResult, RangeType[TypeType]]
 
-    def __init__(self, constant_types: Iterable[TypeType] = ()) -> None:
+    def __init__(self, constant_types: Iterable[Attribute] | None = None) -> None:
```

Comment:
Why not make the caller construct the `ArrayAttr` instead of doing it in this function?